bin/check-unitybegin

```diff
@@ -9,11 +16,35 @@ set +e
 
 ERROR_COUNT=0
 
-for f in exercises/practice/*/test_*; do
-    if ! grep -q UnityBegin\(\""$(basename "$f")"\"\) "$f"; then
-        echo "$f needs correct UnityBegin line"
-        (( ERROR_COUNT++ ))
-    fi;
-done
+check_exercises () {
+    for f in "${1}"/*/test_*; do
+        if [ -f "${f}" ]
+        then
+            if ! grep -q UnityBegin\(\""$(basename "$f")"\"\) "$f"; then
+                echo "$f needs correct UnityBegin line"
+                (( ERROR_COUNT++ ))
+            fi
+        else
+            echo "WARNING: file not found. Ignoring ${1}"
```

Comment:
Im not too sure if it is a good idea to leave this as a warning.
Maybe it should be an error.
It could mean that there is a exercises/concept directory but no concept exercises therein, but it could also mean that there is a concept exercise without the needed files within it.
But then - separation of concerns - should this script care about that?

Reply:
I'm in favor of making this an error. It's likely that it would get caught long before this, but I can't see any reason why we wouldn't take this opportunity to call attention to something unexpected.